docs/setup/k8s/running-in-production-k8s.md

```diff
@@ -0,0 +1,134 @@
+
+Here you will find more advanced configuration options, and some practices which address the needs of running Nuclio in production environments. 
+
+
+#### In this document
+
+- [Preferred installation method](#preferred-installation-method)
+- [Version freezing](#version-freezing)
+- [Multi Tenancy](#multi-tenancy)
+- [Air gapped (dark site) operation](#air-gapped-dark-site-operation)
+- [Using Kaniko as an image builder](#using-kaniko-as-an-image-builder)
+
+
+## Preferred installation method
+
+Even though you may, of course, use any of the available methods, we find that heavy-lifting over k8s is best done with helm charts.
+Nuclio's [helm chart](/hack/k8s/helm/nuclio/) is our preferred mode of deploying Nuclio in [Iguazio](https://www.iguazio.com/) as of writing this document.
+This is also where you'll find most of the production oriented features appearing first, and is the most tightly maintained form of deployment.
+
+Below is a quick example of how to setup the a specific stable version of nuclio (1.3.14) via helm charts:
+
+- Create a namespace for your functions, and a secret with your intended registry's credentials
+
+    ```sh
+    kubectl create namespace nuclio
+    ```
+
+    ```sh
+    read -s mypassword
+    <enter your password>
+    
+    kubectl create secret docker-registry registry-credentials \
+        --docker-username <username> \
+        --docker-password $mypassword \
+        --docker-server <URL> \
+        --docker-email <some email>
+    
+    unset mypassword
+    ```
+
+- Copy the secret to the nuclio namespace, since k8s does not allow for secret sharing between namespaces:
+    ```sh
+    kubectl get secret registry-credentials -n default -o yaml \
+    | sed s/"namespace: default"/"namespace: nuclio"/ \
+    | kubectl apply -f -
+    ```
+
+ - Checkout the nuclio project and install nuclio from its helm chart: 
+
+    ```sh
+    git checkout https://github.com/nuclio/nuclio.git
+    
+    helm install \
+        --set registry.secretName=registry-credentials \
+        --set registry.pushPullUrl=<your registry URL> \
+        --set controller.image.tag=<version>-amd64 \
+        --set dashboard.image.tag=<version>-amd64 \
+        ./hack/k8s/helm/nuclio/
+    ```
+
+  See [the helm chart's values file](/hack/k8s/helm/nuclio/values.yaml) for a full list of configurable parameters
+
+
+## Multi Tenancy
+
+Implementation of multi-tenancy can be done in many different forms and to various degrees. Our experience have led us to adopt the k8s approach of tenant isolation using namespaces.
+- To achieve tenant separation for various nuclio projects and functions, and to avoid cross-tenant contamination and resource races, we have opted to deploy in each namespace a fully functioning Nuclio deployment, and configure Nuclio's controller to be namespaced.
```

Comment:
don't like the the reference to 'contamination' 
it has very negative associations nowdays
suggest changing to 'mixing'

Reply:
In the case of multi tenancy, contamination is the correct term I believe 🤖 but I see what you're saying :|